solidity/contracts/system/TBTCSystem.sol

```diff
@@ -440,7 +441,7 @@ contract TBTCSystem is Ownable, ITBTCSystem, DepositLog {
         view
         returns (uint256)
     {
-        IBondedECDSAKeepFactory _keepFactory = keepFactoryStrategy.chooseFactory(address(this));
+        IBondedECDSAKeepFactory _keepFactory = keepFactoryStrategy.selectFactory(address(this));
```

Comment:
I'm not sure we're able to do it this way. The factory that is going to be selected when actually doing a deposit may change, so it would be safest to get the `openKeepFeeEstimate()` through the selection strategy, querying both and taking the maximum if necessary.

Another way to do it would be to have `selectFactory()` return a constant value until it's refreshed with `selectFactoryAndRefresh()` (which returns the value of `selectFactory()` and also performs the selection of the next factory with a new seed). In that case this would work, but we'd need to `selectFactoryAndRefresh()` in `requestNewKeep()`.

Reply:
I like the `selectFactoryAndRefresh` idea. It's cheaper and querying both will result in a question of what should we do with the surplus.